queue/queue.go

```diff
@@ -78,13 +79,15 @@ func (db *Store) Next(ctx context.Context, resp mdm.Response) (*Command, error)
 	}
 
 	// pop the first command from the queue and add it to the end.
+	// If the regular queue is empty, send a command that got 
```

Comment:
Any thoughts on sending the NotNow command before the rest of the queue? In other words, "pick up where we left off", rather than push NotNow commands to the back of the line.

Reply:
That is a really good question. As long as the device stays connected to APNS and the internet all the time, it should not be a problem and is probably the preferred order. 

a) You want the last command sent to have a Status: NotNow response so that the device polls later.
b) When the device is sleeping, new commands should be preferred because those in the NotNow queue would probably be refused again. If you have both queues filled, it would take twice as much time to empty the Commands queue.
c) Because the device keeps clearing the Commands queue, when it wakes up it will mostly get the NotNow queue anyway.
d) Some of the commands that will never get refused with NotNow are rather high priority, which gives the NotNow queue a slightly lower priority via association.

That logic mostly breaks down if the device first sleeps and then shuts down, so that both queues are filled when it reconnects.